bincopy.py

```diff
@@ -937,6 +1015,39 @@ def i8hex(address):
 
         return '\n'.join(data_address + footer) + '\n'
 
+    def as_ti_txt(self):
+        """Format the binary file as TI-TXT records and return them as a
+        string.
+
+        :returns: A string of TI-TXT records separated by a
+                  newline.
+
+        """
+        lines = []
+
+        def chunks(data, size):
+            """
+            Iterate through data in chunks
+            :param data: data to iterate through
+            :param size: chunk size
+            :return: chunks. Note: Last chunk may be smaller
+            """
+
+            while data:
+                yield data[:size]
+                data = data[size:]
+
+        for segment in self._segments:
+            lines.append("@{:04X}".format(segment.address))
+
+            for chunk in chunks(segment.data, TI_TXT_BYTES_PER_LINE):
+                lines.append(" ".join("{:02X}".format(byte) for byte in chunk))
+
+        lines += "q"
+
+        return "\n".join(lines) + "\n"
```

Comment:
It seems one often want to iterate over chunks and at the same time know if the chunk is at the beginning of a segment. Currently `self._segments.chunks()` does only return the chunk address and its data. I think the chunks-method should take another optional argument, say `offset_into_segment=False`, which if true makes the chunks method return a three-tuple, `address`, `data` and `offset_into_segment`, instead of the two-tuple `address` and `data`. With that extension the chunks method in this function could be replaced with the common one.

Reply:
I was thinking the same, but I think it's better to leave it for you to refactor since you're of course more familiar with the design of the library.

I was also thinking that the rather basic operation of splitting a sequence into chunks is way too difficult to do in Python. Even itertools doesn't seem to support a good way to do it.

Reply:
Yeah, it's quite surprising that Python does not have better support for this operation.

I'll update the chunks method later.